Module options | Transliteration no longer works in xiphos.exe 4.3.1 (gtk2 webkit1)

Module options | **Transliteration** no longer works in **xiphos.exe 4.3.1** (gtk2 webkit1)

Moreover, applying this option to an **Ancient Hebrew** module causes the direction to switch from **RtoL** to **LtoR**.

What went wrong?

Am I the first to test whether this feature still works?

I am attempting to do docker things. I'm not used to containers, so I'm stumbling a bit.

First, it's a certainty that the problem is lack of ICU in Sword.
`diatheke -b LXX -f plain -o t -k g.1.1`
returns Greek, as if `-o t` were not present. By comparison, in Linux:
```
diatheke -b LXX -f plain -k g.1.1
Genesis 1:1: εν αρχη εποιησεν ο θεος τον ουρανον και την γην
diatheke -b LXX -f plain -o t -k g.1.1
Genesis 1:1: en archē epoiēsen o theos ton ouranon kai tēn gēn
```
Second, what I've realized is that the resulting Sword configuration lists a directory for ICU purposes, which doesn't include what's needed. There is no `icu-config`, for example, as I see in the output in Linux build.

Third, there is ICUSWORD. It shows up in config output as "no" but then `build-sword.sh` explicitly sets `-D_ICUSWORD_` anyhow. Just to test, I added `-D_ICU_`, which then caused collapse at the point of linking libsword:
```
.libs/swmgr.o:swmgr.cpp:(.text+0x534c): undefined reference to `sword::UTF8Transliterator::UTF8Transliterator()'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x165): undefined reference to `u_strFromUTF8_57'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x199): undefined reference to `u_strToUpper_57'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x1c9): undefined reference to `u_strToUTF8_57'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x275): undefined reference to `u_strFromUTF8_57'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x2a9): undefined reference to `u_strToLower_57'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x2d9): undefined reference to `u_strToUTF8_57'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x30b): undefined reference to `u_isupper_57'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x32b): undefined reference to `u_islower_57'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x34b): undefined reference to `u_isdigit_57'
.libs/stringmgr.o:stringmgr.cpp:(.text+0x36b): undefined reference to `u_isalpha_57'
.libs/scsuutf8.o:scsuutf8.cpp:(.text+0x38): undefined reference to `ucnv_close_57'
.libs/scsuutf8.o:scsuutf8.cpp:(.text+0x43): undefined reference to `ucnv_close_57'
.libs/scsuutf8.o:scsuutf8.cpp:(.text+0x98): undefined reference to `ucnv_close_57'
.libs/scsuutf8.o:scsuutf8.cpp:(.text+0xa3): undefined reference to `ucnv_close_57'
.libs/scsuutf8.o:scsuutf8.cpp:(.text+0xf3): undefined reference to `ucnv_open_57'
.libs/scsuutf8.o:scsuutf8.cpp:(.text+0x106): undefined reference to `ucnv_open_57'
.libs/scsuutf8.o:scsuutf8.cpp:(.text+0x1a9): undefined reference to `icu_57::UnicodeString::UnicodeString(char const*, int, UConverter*, UErrorCode&)'
.libs/scsuutf8.o:scsuutf8.cpp:(.text+0x1f0): undefined reference to `icu_57::UnicodeString::extract(char*, int, UConverter*, UErrorCode&) const'
.libs/scsuutf8.o:scsuutf8.cpp:(.text+0x2c7): undefined reference to `icu_57::UnicodeString::extract(char*, int, UConverter*, UErrorCode&) const'
```
And so on and so on. Clearly ICU is not recognized via `--with-icu=/usr/${arch}-w64-mingw32/sys-root/mingw`

@greg-hellings, can you offer insight about how ICU could be better detected, recognized, specified? There do exist pkgconfig files in `/usr/x86_64-w64-mingw32/sys-root/mingw/lib/pkgconfig/*icu*` but there is no `icu-config` under .../bin.

--- a/win32/build_sword.sh
+++ b/win32/build_sword.sh
@@ -9,6 +9,7 @@ tar xzf sword.tar.gz
 cd sword-1.9.0
 # It confuses gcc when you have this
 sed -i configure.ac -e 's#-no-undefined##'
+sed -i configure.ac -e 's#with_icu=no#PKG_CHECK_MODULES([ICU],[icu-uc icu-io icu-i18n],false,true)\nAM_CFLAGS="$AM_CFLAGS -D_ICU_"\nAM_CXXFLAGS="$AM_CXXFLAGS -D_ICU_"\nLIBS="$LIBS $ICU_LIBS"#'
 # But ld NEEDS to have it for the shared library
 sed -i lib/Makefile.am -E -e 's#libsword_la_LDFLAGS =(.*)#libsword_la_LDFLAGS = -no-undefined \1#'
 autoreconf
@@ -23,7 +24,7 @@ function build_sword {
     export RANLIB="${arch}-w64-mingw32-ranlib"
     export STRIP="${arch}-w64-mingw32-strip"
     export LD="${arch}-w64-mingw32-ld"
-    export CFLAGS="-I/usr/${arch}-w64-mingw32/sys-root/mingw/include -I$(pwd)/../sword-1.9.0/include -D_ICUSWORD_"
+    export CFLAGS="-I/usr/${arch}-w64-mingw32/sys-root/mingw/include -I$(pwd)/../sword-1.9.0/include"
     export CXXFLAGS="${CXXFLAGS} ${CFLAGS}"
     export LDFLAGS="-L/usr/${arch}-w64-mingw32/sys-root/mingw/lib"
     export PATH="/usr/${arch}-w64-mingw32/sys-root/mingw/bin:${PATH}"
@@ -33,6 +34,7 @@ function build_sword {
     ../sword-1.9.0/configure \
         --enable-shared \
         --disable-static \
+        --with-icusword \
         --host=${arch}-w64-mingw32 \
         --target=${arch}-w64-mingw32 \
         --build=${arch}-linux-gnu \
@@ -50,9 +52,8 @@ function build_sword {
         --sharedstatedir=/usr/${arch}-w64-mingw32/sys-root/mingw/com \
         --mandir=/usr/${arch}-w64-mingw32/sys-root/mingw/share/man \
         --infodir=/usr/${arch}-w64-mingw32/sys-root/mingw/share/info \
-        --with-clucene=/usr/${arch}-w64-mingw32/sys-root/mingw \
-        --with-icu=/usr/${arch}-w64-mingw32/sys-root/mingw || cat config.log
-    make -j10
+        --with-clucene=/usr/${arch}-w64-mingw32/sys-root/mingw || cat config.log
+    make -j10 || cat config.log
     make install
     mv "utilities/.libs/addld.exe" "/usr/${arch}-w64-mingw32/sys-root/mingw/bin/addld.exe"
     cd ..

